### PR TITLE
STM32U0 Support

### PR DESCRIPTION
**Pull request by FranzForstmayr**
PR to enable STM32U0

### CODE REVIEW

File: tools/generator/dfg/stm32/stm.py [hunk 1]
```diff
@@ -778,6 +784,28 @@ def getDmaRemap(did, dma, channel, driver, inst, signal):
...
+                'name': ['31'],
+                'memories': {'flash': 0, 'sram1': 2*1024, 'sram2': 1*1024}
+            },
+            {
+                'name': ['73'],
+                'memories': {'flash': 0, 'sram1': 2*1024, 'sram2': 1*1024}
+            },
+            {
+                'name': ['83'],
+                'memories': {'flash': 0, 'sram1': 2*1024, 'sram2': 1*1024}
```
Comment (salkinium) [inline tools/generator/dfg/stm32/stm.py:796]: can all be merged into one line

File: tools/generator/dfg/stm32/stm_groups.py [hunk 2]
```diff
@@ -378,6 +378,20 @@
...
+        'family': ['u0'],
+        'name': ['31']
+    },
+    {
+        'family': ['u0'],
+        'name': ['73']
+    },
+    {
+        'family': ['u0'],
+        'name': ['83']
```
Comment (salkinium) [inline tools/generator/dfg/stm32/stm_groups.py]: 73 and 83 can probably be merged

File: tools/generator/Makefile [hunk 3]
```diff
@@ -4,7 +4,7 @@ ext/%:
 	@git clone https://github.com/modm-io/$(@:ext/%=%).git $@
 
 ext/stm32-cube-hal-drivers:
-	@git clone https://github.com/modm-ext/stm32-cube-hal-drivers.git $@
+	@git clone -b updater_git https://github.com/FranzForstmayr/stm32-cube-hal-drivers.git $@
```
Comment (salkinium) [inline tools/generator/Makefile]: should be fixed now

File: tools/generator/dfg/stm32/stm.py [hunk 4]
```diff
@@ -8,6 +8,8 @@
 
 ignored_devices = \
 [
+    "STM32C071",
+    "STM32G414",
```
Comment (salkinium) [inline tools/generator/dfg/stm32/stm.py:12]: what's missing for these?

**Review by salkinium — COMMENTED** [review]
Thanks!

**Comment by salkinium** [comment]
I also just realized that I seem to have missed that the CI has been broken for two months… whoops.

File: tools/generator/dfg/stm32/stm.py (hunk 4 above)
Comment (FranzForstmayr) [reply tools/generator/dfg/stm32/stm.py:12]: The Die file is missing in the memory subdirectory.

**Comment by FranzForstmayr** [comment]
Hmm CI is broken now due to the merging of STM32U073 and STM32U083, is there a way to clear caches?

**Review by salkinium — APPROVED** [review]
Thanks!

**Pull request merged**